### PR TITLE
Return original path 

### PR DESCRIPTION
**Pull request by updateaman**
To fix issue #400 
Return original directory path passed by the user from MockDirectoryInfo ToString() method.

Which is what the original directoryInfo class does:
https://github.com/dotnet/corefx/blob/master/src/System.IO.FileSystem/src/System/IO/DirectoryInfo.cs

### CODE REVIEW

**Comment by fgreinacher** [comment]
Thanks for working on this @updateaman - the changes look good so far. Could you additionally check if we have the same inconsistency for `FileInfo` and `DriveInfo`?

**Comment by updateaman** [comment]
@fgreinacher yes, they had the same problem with ToString(). I've fixed them and added tests

File: System.IO.Abstractions.TestingHelpers.Tests/MockDriveInfoTests.cs [hunk 1]
```diff
@@ -59,13 +59,32 @@ public void MockDriveInfo_RootDirectory_ShouldReturnTheDirectoryBase()
...
+        [TestCase("e:","e:\\")]
+        [TestCase("f:","f:\\")]
+        public void MockDriveInfo_ToString_ShouldReturnTheDrivePath(string path, string expectedPath)
+        {
+            // Arrange
+            var directoryPath = XFS.Path(path);
+            
+            // Act
+            var mockDriveInfo = new MockDriveInfo(new MockFileSystem(), directoryPath);
+            var realDriveInfo = new DriveInfo(directoryPath);
```
Comment (fgreinacher) [inline System.IO.Abstractions.TestingHelpers.Tests/MockDriveInfoTests.cs]: We usually don't "compare" against real implementations, so please remove this one

File: System.IO.Abstractions.TestingHelpers.Tests/MockFileInfoTests.cs [hunk 2]
```diff
@@ -491,6 +491,24 @@ public void MockFileInfo_CopyTo_ThrowsExceptionIfSourceDoesntExist()
...
+        [TestCase(@"c:\a\c.txt")]
+        [TestCase(@"c:\c.txt")]
+        public void MockFileInfo_ToString_ShouldReturnOriginalFilePath(string path)
+        {
+            //Arrange
+            var filePath = XFS.Path(path);
+
+            //Act
+            var mockFileInfo = new MockFileInfo(new MockFileSystem(), filePath);
+            var realFileInfo = new FileInfo(filePath);
```
Comment (fgreinacher) [inline System.IO.Abstractions.TestingHelpers.Tests/MockFileInfoTests.cs]: Same comment as above, please remove this

File: System.IO.Abstractions/DriveInfoBase.cs [hunk 3]
```diff
@@ -142,5 +142,12 @@ protected DriveInfoBase(IFileSystem fileSystem)
 
             return new DriveInfoWrapper(new FileSystem(), driveInfo);
         }
+
+        /// <summary>Returns a drive name as a string.</summary>
+        /// <returns>The name of the drive.</returns>
+        public override string ToString()
```
Comment (fgreinacher) [inline System.IO.Abstractions/DriveInfoBase.cs]: To reduce public API changes and therefore downstream churn please undo the changes in `DriveInfoBase` and override `ToString` in `DriveInfoWrapper`/`MockDriveInfo` instead.

File: System.IO.Abstractions/DirectoryInfoBase.cs [hunk 4]
```diff
@@ -7,7 +7,7 @@ namespace System.IO.Abstractions
     [Serializable]
     public abstract class DirectoryInfoBase : FileSystemInfoBase
     {
-        protected DirectoryInfoBase(IFileSystem fileSystem) : base(fileSystem)
+        protected DirectoryInfoBase(IFileSystem fileSystem, string path) : base(fileSystem, path)
```
Comment (fgreinacher) [inline System.IO.Abstractions/DirectoryInfoBase.cs]: To reduce public API changes and therefore downstream churn please undo the changes in this file and override `ToString` in `*Wrapper`/`Mock*` instead.

File: System.IO.Abstractions/FileInfoBase.cs [hunk 5]
```diff
@@ -6,7 +6,7 @@ namespace System.IO.Abstractions
     [Serializable]
     public abstract class FileInfoBase : FileSystemInfoBase
     {
-        protected FileInfoBase(IFileSystem fileSystem) : base(fileSystem)
+        protected FileInfoBase(IFileSystem fileSystem, string path) : base(fileSystem, path)
```
Comment (fgreinacher) [inline System.IO.Abstractions/FileInfoBase.cs]: To reduce public API changes and therefore downstream churn please undo the changes in this file and override `ToString` in `*Wrapper`/`Mock*` instead.

File: System.IO.Abstractions/FileSystemInfoBase.cs [hunk 6]
```diff
@@ -4,9 +4,12 @@
     [Serializable]
     public abstract class FileSystemInfoBase
     {
-        protected FileSystemInfoBase(IFileSystem fileSystem)
+        private readonly string originalPath;
+
+        protected FileSystemInfoBase(IFileSystem fileSystem, string path)
```
Comment (fgreinacher) [inline System.IO.Abstractions/FileSystemInfoBase.cs]: To reduce public API changes and therefore downstream churn please undo the changes in this file and override `ToString` in `*Wrapper`/`Mock*` instead.

File: System.IO.Abstractions.TestingHelpers.Tests/MockDriveInfoTests.cs (hunk 1 above)
Comment (updateaman) [reply System.IO.Abstractions.TestingHelpers.Tests/MockDriveInfoTests.cs]: Removed real implementation in test

File: System.IO.Abstractions/DriveInfoBase.cs (hunk 3 above)
Comment (updateaman) [reply System.IO.Abstractions/DriveInfoBase.cs]: Removed real implementation in test

File: System.IO.Abstractions.TestingHelpers.Tests/MockFileInfoTests.cs (hunk 2 above)
Comment (updateaman) [reply System.IO.Abstractions.TestingHelpers.Tests/MockFileInfoTests.cs]: Removed real implementation in test

File: System.IO.Abstractions/DriveInfoBase.cs (hunk 3 above)
Comment (updateaman) [reply System.IO.Abstractions/DriveInfoBase.cs]: ToString moved to MockDriveInfo class

File: System.IO.Abstractions/DirectoryInfoBase.cs (hunk 4 above)
Comment (updateaman) [reply System.IO.Abstractions/DirectoryInfoBase.cs]: ToString moved to MockDirectoryInfo

File: System.IO.Abstractions/FileInfoBase.cs (hunk 5 above)
Comment (updateaman) [reply System.IO.Abstractions/FileInfoBase.cs]: ToString moved to MockFileInfo

File: System.IO.Abstractions/FileSystemInfoBase.cs (hunk 6 above)
Comment (updateaman) [reply System.IO.Abstractions/FileSystemInfoBase.cs]: ToString removed from FileSystemInfoBase

**Comment by fgreinacher** [comment]
LGTM, thanks again @updateaman!
